src/v/kafka/server/connection_context.h

```diff
@@ -131,26 +136,46 @@ class connection_context final
           name, operation, principal, security::acl_host(_client_addr));
 
         if (!authorized) {
-            if (quiet) {
-                vlog(
-                  _authlog.debug,
-                  "proto: {}, sasl state: {}, acl op: {}, principal: {}, "
-                  "resource: {}",
-                  _proto.name(),
-                  security::sasl_state_to_str(_sasl.state()),
-                  operation,
-                  principal,
-                  name);
+            if (_sasl) {
+                if (quiet) {
```

Comment:
nit: i'd move this logging out to a separate function. It kinda gets in the way of the logic.